src/huggingface_hub/hf_api.py

```diff
@@ -633,7 +633,7 @@ def list_models(
         Get the public list of all the models on huggingface.co
 
         Args:
-            filter (:class:`ModelFilter` or :obj:`str` or :class:`Iterable`, *optional*):
+            filter (`ModelFilter` or `str` or `Iterable`, *optional*):
```

Comment:
In the rendered version `ModelFilter` seems to be a simple text and not a hyperlink to `ModelFilter`'s documentation. I think we should keep those link with this migration (`:class:` does that in rst)